pkg/ocm/config.go

```diff
@@ -50,12 +98,20 @@ func GetEnv() (string, error) {
 		urlAliases = fedramp.URLAliases
 	}
 
+	// Check for URL aliases
 	for env, api := range urlAliases {
 		if api == strings.TrimSuffix(cfg.URL, "/") {
 			return env, nil
 		}
 	}
 
+	// In case of regionalized URLs, check for env aliases
+	for env, api := range EnvAliases {
```

Comment:
no. we already solved this when determining what global gateway to query for ocm-shards.json: https://github.com/openshift-online/ocm-sdk-go/blob/main/rh_region.go#L48

Its the same thing here, we just need an "env alias" instead of the gateway url. heres your pseudo code:
```
//
discoveryGatewayUrl, _ := URL.Parse(DetermineRegionDiscoveryUrl(cfg.URL))
for env, api := range EnvAlias {
  if strings.Contains(api, discoveryGatewayUrl.Host){
    return env, nil
  }
}
```

Reply:
hmmm. actually i take that back @tirthct , i think what you have is fine actually. Nice job. Sorry, pre-coffee :)

Reply:
actually i retract my retraction. i really need coffee before i code review. 

We need to use `DetermineRegionDiscoveryUrl` to normalize cfg.URL to the discorvery url to handle the production use case.:
```
cfg.URL=api.aws.ap-southeast-1.stage.openshift.com
urlAliases = {"stage" : "api.stage.openshift.com"}
api.aws.ap-southeast-1.stage.openshift.com => https://api.stage.openshift.com/static/ocm-shards.json => api.stage.openshift.com => stage

cfg.URL=api.aws.ap-southeast-1.openshift.com
urlAliases = {"prod" : "api.openshift.com"}
api.aws.ap-southeast-1.openshift.com => https://api.openshift.com/static/ocm-shards.json => api.openshift.com => prod
```

Reply:
As per the discussion in https://redhat-internal.slack.com/archives/C0421HEGLSW/p1709230320256999, fixed the logic